### DISCUSSION TITLE
How should we configure the guest agent service listener address

### DISCUSSION THREAD

**Discussion by s0nea** [rancher-sandbox/rancher-desktop#3803] [Research]
### Problem Description

https://github.com/rancher-sandbox/rancher-desktop-agent/pull/27 adds a service listener address as an input parameter to the guest agent. It would be nice to reflect the change somewhere in the UI.

### Proposed Solution

I'm not yet sure how the solution should look like. I think the input field could be either part of the first run window or the preferences page. Because I assume the config needs to be rewritten and the service to be restarted, I thought the first run window would be a good idea, e.g.:

![Screenshot_2023-01-17_20-08-22](https://user-images.githubusercontent.com/8761082/213137765-59dd7130-5349-44a5-903f-43dfca6d8823.png)

But there might be reasons for other places I currently don't see.

Also, I was wondering which type of input field to use. At first I thought about adding an IP-address field, e.g.: https://www.npmjs.com/package/vue-ip . But then I noticed the Rancher UI DevKit https://rancher.github.io/dashboard/ doesn't mention an IP address field add all. So I assumed to better stick with an usual Labeled-Input field.

**Comment by rak-phillip** [comment]
I'm open to adding this new field to the first run dialog, but it doesn't look like the right place. 

Our goal with the first run dialog is to keep it as lean as possible so that a user can accept the defaults and start working with Rancher Desktop without any friction. `Guestagent Service Listener Address` sounds like some advanced configuration, and it's not immediately easy to understand how changing this value will affect Rancher Desktop. 

This could fit under preferences, perhaps under Virtual Machine => Network? 

Some questions that come to mind while thinking about adding this to the first run dialog:

- Can I change this value and end up with an invalid configuration?
- Will accepting the default ever leave me with a broken install?
- Will it be obvious what needs to change if I enter an invalid IP?

**Reply by jandubois** [reply]
> This could fit under preferences, perhaps under Virtual Machine => Network?

I just talked with @Nino-K about it (and looked at #3093), and this is actually a Kubernetes and not a VM setting.

**Comment by rak-phillip** [comment]
I'm not seeing any instances of a masked input for IP addresses in Rancher Dashboard. This might be a good enhancement for `@rancher/components`.

**Reply by jandubois** [reply]
> I'm not seeing any instances of a masked input for IP addresses in Rancher Dashboard.

It is my understanding that in this particular use-case the valid entries are limited to `0.0.0.0` and `127.0.0.1` (and maybe the host IP address).

If this is the case, then a dropdown list seems more appropriate than a generic IP address input widget.

**Comment by jandubois** [comment]
> Our goal with the first run dialog is to keep it as lean as possible so that a user can accept the defaults and start working with Rancher Desktop without any friction. `Guestagent Service Listener Address` sounds like some advanced configuration, and it's not immediately easy to understand how changing this value will affect Rancher Desktop.

I agree with this.

Additional, the description is written from our point of view; we don't expect the user to know what the "Guest agent service listener" is, so why should they care about the bind address? Can we find a description that explains what difference it makes to them? Maybe it needs to go into a tooltip?

**Comment by Nino-K** [comment]
In addition to what is mentioned above, why the first dialog box may not be a suitable place for this is, users would have to perform a factory reset and this config change should not require a factory reset. Maybe just restarting Rancher Desktop would be adequate since that also restarts `rancher-desktop-guestagent` service?

Also, another option to incorporate this configuration is via provisioning script. This needs to be backed by documentation where we elaborate on what it does and what are the available values for the property to avoid user misconfiguration.

**Comment by jandubois** [comment]
> In addition to what is mentioned above, why the first dialog box may not be a suitable place for this is, users would have to perform a factory reset

All settings from the first-run dialog are changeable via the regular Preferences dialog as well, so that would not be an issue.

But as @rak-phillip already mentioned, the first-run dialog should only prompt for essential settings that virtually every user wants to setup correctly from the start, and that's why this option doesn't qualify.

To start with, it is still not clear to me what exactly it is that it configures. 😄

**Comment by Nino-K** [comment]
Just going to attach the original issue here: https://github.com/rancher-sandbox/rancher-desktop/issues/3093

**Comment by s0nea** [comment]
We agreed on integrating the setting as a CLI command and not as part of the UI for now: https://github.com/rancher-sandbox/rancher-desktop/issues/3940
